pkg/controller/jobframework/integrationmanager_test.go

```diff
@@ -0,0 +1,259 @@
+			integrationName: "newFramework",
+			integrationCallbacks: IntegrationCallbacks{
+				SetupWebhook: testSetupWebhook,
+				WebhookType:  testWebhookType,
+				SetupIndexes: testSetupIndexes,
+				AddToScheme:  testAddToScheme,
+			},
+			wantError: errMissingMadatoryCallback,
+			wantList:  []string{},
+		},
+		"missing SetupWebhook": {
+			manager:         &integrationManager{},
+			integrationName: "newFramework",
+			integrationCallbacks: IntegrationCallbacks{
+				NewReconciler: testNewReconciler,
+				WebhookType:   testWebhookType,
+				SetupIndexes:  testSetupIndexes,
+				AddToScheme:   testAddToScheme,
+			},
+			wantError: errMissingMadatoryCallback,
+			wantList:  []string{},
+		},
+		"missing WebhookType": {
+			manager:         &integrationManager{},
+			integrationName: "newFramework",
+			integrationCallbacks: IntegrationCallbacks{
+				NewReconciler: testNewReconciler,
+				SetupWebhook:  testSetupWebhook,
+				SetupIndexes:  testSetupIndexes,
+				AddToScheme:   testAddToScheme,
+			},
+			wantError: errMissingMadatoryCallback,
+			wantList:  []string{},
+		},
+		"missing SetupIndexes": {
+			manager:         &integrationManager{},
+			integrationName: "newFramework",
+			integrationCallbacks: IntegrationCallbacks{
+				NewReconciler: testNewReconciler,
+				SetupWebhook:  testSetupWebhook,
+				WebhookType:   testWebhookType,
+				AddToScheme:   testAddToScheme,
+			},
+			wantError: nil,
+			wantList:  []string{"newFramework"},
+			wantCallbacks: IntegrationCallbacks{
+				NewReconciler: testNewReconciler,
+				SetupWebhook:  testSetupWebhook,
+				WebhookType:   testWebhookType,
+				AddToScheme:   testAddToScheme,
+			},
+		},
+		"missing AddToScheme": {
+			manager:         &integrationManager{},
+			integrationName: "newFramework",
+			integrationCallbacks: IntegrationCallbacks{
+				NewReconciler: testNewReconciler,
+				SetupWebhook:  testSetupWebhook,
+				WebhookType:   testWebhookType,
+				SetupIndexes:  testSetupIndexes,
+			},
+			wantError: nil,
+			wantList:  []string{"newFramework"},
+			wantCallbacks: IntegrationCallbacks{
+				NewReconciler: testNewReconciler,
+				SetupWebhook:  testSetupWebhook,
+				WebhookType:   testWebhookType,
+				SetupIndexes:  testSetupIndexes,
+			},
+		},
+	}
+
+	for tcName, tc := range cases {
+		t.Run(tcName, func(t *testing.T) {
+			gotError := tc.manager.register(tc.integrationName, tc.integrationCallbacks)
+			if diff := cmp.Diff(tc.wantError, gotError, cmpopts.EquateErrors()); diff != "" {
+				t.Errorf("Unexpected error (-want +got):\n%s", diff)
+			}
+			gotList := tc.manager.getList()
+			if diff := cmp.Diff(tc.wantList, gotList); diff != "" {
+				t.Errorf("Unexpected frameworks list (-want +got):\n%s", diff)
+			}
+
+			if gotCallbacks, found := tc.manager.get(tc.integrationName); found {
+				if diff := cmp.Diff(tc.wantCallbacks, gotCallbacks, cmp.FilterValues(func(_, _ interface{}) bool { return true }, cmp.Comparer(compareCallbacks))); diff != "" {
```

Comment:
Is `FilterValues` necessary?

Reply:
Actually, we are not gaining anything by doing `cmp.Diff` here, given that we are comparing each field individually in `compareCallbacks`.

I would just do `tc.wantCallbacks.NewReconciler != gotCallBacks.NewReconciler` and similarly for other fields

Reply:
it adds some consistency in the error reporting

Reply:
unfortunately yes